### DISCUSSION TITLE
v3.1.0 Released

### DISCUSSION THREAD

**Discussion by coolapso** [streamdeck-linux-gui/streamdeck-linux-gui#53] [Announcements]
Hi everyone, 

Version 3.1.0 has just been released, release notes are available here: https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/releases/tag/v3.1.0

Thank you a lot to everyone who contributed!

**Comment by oparada1988** [comment]
Dumb question, how do we go about updating? Sorry, kind of new,

pipx upgrade streamdeck-linux-gui
streamdeck-linux-gui is already at latest version 3.0.1 (location:
/home/oscar/.local/pipx/venvs/streamdeck-linux-gui)

**Reply by coolapso** [reply]
you should be able to upgrade just like any other package with ~~pip~~ pipx. the new version is definitely on pypi https://pypi.org/project/streamdeck-linux-gui/ 

Edit: Redacted pip, meant pipx

**Comment by dhtseany** [comment]
Hello, has there been any dependency changes for this release? Thanks!

**Reply by coolapso** [reply]
Some dependencies were removed but they are related with development not so much with the application itself.

**Comment by dhtseany** [comment]
Hi, I'm the maintainer of the non-pip package on Arch. I need to know what
package dep's you dropped so I can remove them from the requirements
list. Those
changes should be listed on your release page
<https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/releases/tag/v3.1.0>.
Thanks, Sean

**Reply by coolapso** [reply]
Sorry will try to make a better job regarding dependencies next time.

I am well aware of the AUR package, and will be always very keen notifying of anything that might affect it :) 

None of the changes should Affect the AUR package as most of them were dev dependencies related with the project scaffolding, and unit testing. 

Here goes a list of some dependency changes:

python updated from => 3.7 to =>3.8

Dropped: 

Portray
hypothesis-auto


Added: 

mkdocs = ">=1.4.2"
mkdocs-material = ">=9.1.14"

**Comment by oparada1988** [comment]
> you should be able to upgrade just like any other package with pip. the new version is definitely on pypi https://pypi.org/project/streamdeck-linux-gui/

Not sure if I am doing something wrong on my end, I went ahead and uninstall it and then reinstalled it. Went over all of the steps as per the ubuntu guide:

![Screenshot from 2023-08-27 12-51-45](https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/assets/4031078/bddd7df3-5416-4bf2-bdcf-93ab7b96b412)

I then uninstalled it again and tried "pip install streamdeck-linux-gui" and got the following:

![Screenshot from 2023-08-27 12-53-09](https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/assets/4031078/7b789109-0479-4394-a406-4f44b090be45)


I am using ubuntu 22.04.3 LTS. Please let me know if I am doing something wrong on my end

**Reply by coolapso** [reply]
Definitely loading the wrong version of the package and most likely something wrong on your side. 

what do you get with: 

`sudo find / -name streamdeck` ? 

As I don't really have a reliable way to determine what is wrong, 
i'd recommend you to backup your streamdeck config and make a clean install

**Reply by oparada1988** [reply]
I went ahead and reinstalled it using "python3 -m pipx install streamdeck-linux-gui" But it is still installing 3.0.1

![Screenshot from 2023-08-27 13-28-17](https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/assets/4031078/c5f5d0d4-85b7-44d4-811d-58b323732296)

**Reply by oparada1988** [reply]
IDK if im being stupid, but I just went ahead and downloaded the built files and overwrote the ones in "~/.local/lib/python3.10/site-packages" and i got the new release running now. is there a way I can check if pip is pointed to the correct url? IDK, just throwing thoughts out there
![Screenshot from 2023-08-27 13-43-55](https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/assets/4031078/66be59c0-c81e-468d-b2b5-cb90fd940237)

**Reply by oparada1988** [reply]
Hi, I made a video for your review. I completely uninstalled streamdeck and deleted all files. when doing pipx install i still get 3.0.1:

https://youtu.be/cadaTTOxl_Q?si=HqwN0FyFMJFKy6Hr

**Comment by dhtseany** [comment]
Perfect, I thank you and your team for your efforts picking up this project
and doing your best to keep it moving forward. Thanks, Sean

**Reply by coolapso** [reply]
I accidentally posted that message before actually having it finished, updated it above ;)

**Comment by Narcodepsy** [comment]
Email response to wrong bug. Ignore

**Reply by jonathanhirschman** [reply]
Having exactly the same issue - pipx wants to install an old version. See:
![image](https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/assets/33166581/ddb4d019-a42b-44ce-acd9-8cbbf78d9b8a)

Also on Ubuntu (22.04) - perhaps this is an issue particular to this distribution and/or version?
